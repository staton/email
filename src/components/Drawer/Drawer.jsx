import React, { Component } from 'react';
import DrawerMenu from '../DrawerMenu/DrawerMenu';
import ScreenOverlay from '../ScreenOverlay/ScreenOverlay';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { setDrawerVisibility } from '../../redux/actions/app-actions';

export class Drawer extends Component {

    constructor() {
        super();

        this.handleDrawerShadowClicked = this.handleDrawerShadowClicked.bind(this);
    }

	render() {
		return (
			<div className="Drawer">
                <div className={this.getDrawerContentClassNames()}>
                    <DrawerMenu />
                </div>
                {
                    // The shadow that appears behind the drawer should only be visible on small screens
                    (this.props.isSmallScreen)
                    ?   <ScreenOverlay
                            isVisible={this.props.isDrawerVisible}
                            onClick={this.handleDrawerShadowClicked}
                            zIndex={78}
                        />
                    :   null
                }
			</div>
		);
    }

    /**
     * Gets the inline style for the drawer. The style for the drawer will be updated
     * if the isDrawerVisible prop is true.
     * @returns {string} The name of the drawer content classes.
     */
    getDrawerContentClassNames() {
        const drawerContentClassName = 'Drawer__content';
        const drawerContentHiddenClassNames = drawerContentClassName + ' Drawer__content--hidden';

        return (this.props.isDrawerVisible)
            ? drawerContentClassName
            : drawerContentHiddenClassNames;
    }

    /**
     * Called when the user taps on the drawer shadow.
     * @param {object} e The event.
     */
    handleDrawerShadowClicked(e) {
        this.props.setDrawerVisibility(false);
    }

}

function mapStateToProps(store, ownProps) {
    return {
        isDrawerVisible: store.app.isDrawerVisible,
        isSmallScreen: store.app.isSmallScreen
    };
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({
        setDrawerVisibility: setDrawerVisibility
    },
    dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(Drawer);