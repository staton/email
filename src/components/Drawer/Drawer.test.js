import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import { Drawer } from './Drawer';

describe('<Drawer />', () => {

    it('renders correctly for small screens', () => {
        const wrapper = shallow(
            <Drawer 
                isDrawerVisible={true}
                isSmallScreen={true} 
            />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders correctly for large screens', () => {
        const wrapper = shallow(
            <Drawer 
                isDrawerVisible={true}
                isSmallScreen={false} 
            />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('has classes to handle drawer visibility for small screens when drawer should be visible', () => {
        const wrapper = shallow(
            <Drawer 
                isDrawerVisible={true}
                isSmallScreen={true} 
            />
        );
        expect(wrapper.find('.Drawer__content').length).toEqual(1);
        expect(wrapper.find('.Drawer__content--hidden').length).toEqual(0);
    });

    it('has classes to handle drawer visibility for large screens when drawer should be visible', () => {
        const wrapper = shallow(
            <Drawer 
                isDrawerVisible={true}
                isSmallScreen={false} 
            />
        );
        expect(wrapper.find('.Drawer__content').length).toEqual(1);
        expect(wrapper.find('.Drawer__content--hidden').length).toEqual(0);
    });

    it('does not have classes to handle drawer visibility for small screens when drawer should be hidden', () => {
        const wrapper = shallow(
            <Drawer 
                isDrawerVisible={false}
                isSmallScreen={true} 
            />
        );
        expect(wrapper.find('.Drawer__content').length).toEqual(1);
        expect(wrapper.find('.Drawer__content--hidden').length).toEqual(1);
    });

    it('does not have classes to handle drawer visibility for large screens when drawer should be hidden', () => {
        const wrapper = shallow(
            <Drawer 
                isDrawerVisible={false}
                isSmallScreen={false} 
            />
        );
        expect(wrapper.find('.Drawer__content').length).toEqual(1);
        expect(wrapper.find('.Drawer__content--hidden').length).toEqual(1);
    });

});