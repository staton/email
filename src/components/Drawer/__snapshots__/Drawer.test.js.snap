// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer /> renders correctly for large screens 1`] = `
<div
  className="Drawer"
>
  <div
    className="Drawer__content"
  >
    <Connect(DrawerMenu) />
  </div>
</div>
`;

exports[`<Drawer /> renders correctly for small screens 1`] = `
<div
  className="Drawer"
>
  <div
    className="Drawer__content"
  >
    <Connect(DrawerMenu) />
  </div>
  <ScreenOverlay
    isVisible={true}
    onClick={[Function]}
    zIndex={78}
  />
</div>
`;
